cpu/atmega_common/periph/timer.c

```diff
@@ -158,7 +158,9 @@ int timer_set_absolute(tim_t tim, int channel, unsigned int value)
     unsigned state = irq_disable();
 
     ctx[tim].dev->OCR[channel] = (uint16_t)value;
-    *ctx[tim].flag &= ~(1 << (channel + OCF1A));
+    /* clear spurious IRQs, if any */
+    *ctx[tim].flag |= (1 << (channel + OCF1A));
```

Comment:
```suggestion
    *ctx[tim].flag = (1 << (channel + OCF1A));
```

I think this should also work. @kfessel: You're the AVR expert, do I read the datasheet correctly?

Reply:
In fact, "would also work" is incorrect: The current version would unset all flags and is buggy.

Reply:
i was about to write that:

"Alternatively, <FLAG> can be cleared by writing a logic one to its bit location."

so if you  `flags = flags`  it will clear all flags that are 1.
 
`*ctx[tim].flag |= (1 << (channel + OCF1A));` will clear all flags that are read as 1 and `OCF1A` in any case.

but the compiler should make  `*ctx[tim].flag |= (1 << (channel + OCF1A));` work anyway since all flags are placed in io-address-space  -> it will become `SBI  <FLAGS>, (channel + OCF1A)` writing a one to that specific location